src/manifest.loader.ts

```diff
@@ -2,15 +2,84 @@ import { fsa } from '@linzjs/s3fs';
 import { promises as fs } from 'fs';
 import { logger } from './log';
 import { Manifest, ManifestFile } from './manifest';
+import { hashFile } from './hash';
 
 const BackupExtension = '.1';
+export const ManifestFileName = 'manifest.json';
 
 function isManifestPath(fileName: string): boolean {
   if (fileName.endsWith('.json' + BackupExtension)) return true;
   if (fileName.endsWith('.json')) return true;
   return false;
 }
 
+/**
+ * Compare two manifest.json file hashes. Return true if both hashes match.
+ *
+ * @param manifestA
+ * @param manifestB
+ * @returns
+ */
+async function isSameManifestFileHash(manifestA: ManifestLoader, manifestB: ManifestLoader): Promise<Promise<boolean>> {
+  const hashManifestA = await hashFile(fsa.readStream(manifestA.sourcePath));
+  const hashManifestB = await hashFile(fsa.readStream(manifestB.sourcePath));
+
+  if (hashManifestA === hashManifestB) {
+    // Both manifest.json files have the exact same content.
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Compare the 'Files' data of two manifest.json files. Return true if manifestA contains all the files (with identical data) of ManifestB.
+ * ManifestA can contain more files than ManifestB.
+ *
+ * @param manifestA
+ * @param manifestB
+ * @returns
+ */
+export function isManifestsDifferent(manifestA: ManifestLoader, manifestB: ManifestLoader): boolean {
+  let comparedValue = undefined;
+
+  for (const [key, value] of manifestB.files.entries()) {
+    comparedValue = manifestA.files.get(key);
+
+    if (
+      comparedValue?.path === value.path &&
+      comparedValue?.size === value.size &&
+      comparedValue?.hash === value.hash
+    ) {
+      continue;
+    } else {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * Return true if a manifest.json generated for the same files exists in the destination folder.
+ *
+ * @param currentManifest
+ * @param destPath
+ * @returns
+ */
+export async function isDifferentManifestExists(currentManifest: ManifestLoader, destPath: string): Promise<boolean> {
+  if (!destPath.endsWith('/')) destPath = destPath + '/';
+  const existingManifestPath = fsa.join(destPath, ManifestFileName);
+
+  if (fsa.exists(existingManifestPath)) {
```

Comment:
`fsa.exists` returns a `Promise<boolean>` this will always be true.

```javascript
const f = Promise.resolve(false)
const t = Promise.resolve(true)

if (f) // true
if (t) // also true

if (await f) // false
if (await t) // true
```